src/PairedRouting.php

```diff
@@ -772,6 +772,34 @@ public function add_permalink_settings_notice() {
 		<?php
 	}
 
+	/**
+	 * Update canonical URL for legacy reader paired urls.
+	 *
+	 * @param string $redirect_url Canonical redirect URL.
+	 *
+	 * @return string Updated canonical URL.
+	 */
+	public function maybe_update_paired_url( $redirect_url ) {
```

Comment:
Let's call this method something like `filter_redirect_canonical_to_fix_cpage_requests`